lib/parser/source.cc

```diff
@@ -124,7 +125,13 @@ bool SourceFile::Open(std::string path, std::stringstream *error) {
   errno = 0;
   fileDescriptor_ = open(path.c_str(), O_RDONLY);
   if (fileDescriptor_ < 0) {
-    *error << "could not open " << errorPath << ": " << std::strerror(errno);
+    *error << "Could not open " << errorPath << ": " << std::strerror(errno);
+    return false;
+  }
+  if (isModuleFile_ && flock(fileDescriptor_, LOCK_SH | LOCK_NB) != 0) {
```

Comment:
You could just lock all files, not just modules.

Reply:
_Also, when locking fails, it seems harsh to abort compilation.  Why not try a `sleep(1)` and a few retries?_

Reply:
Locking all files would be okay, except the error would have to be less specific. For .mod files we can say "another compilation is writing it". If it applies to any file, all we can say is "the file is locked" or something.

Reply:
I chose not to retry because this error would normally indicate a failure of the build process. I think it's more helpful to users to know about that than to try to plow through and get their file compiled.